corehq/motech/repeaters/tests/test_tasks.py

```diff
@@ -82,162 +90,221 @@ def form_context(form_ids):
-            state=State.Cancelled,
+            connection_settings=self.conn_settings,
         )
 
-        _process_repeat_record(repeat_record)
-
-        self.assertEqual(self.mock_fire.call_count, 0)
-        self.assertEqual(self.mock_postpone_by.call_count, 0)
-
-    def test_cancels_and_returns_if_domain_cannot_forward(self):
-        self.mock_domain_can_forward.return_value = False
-
         repeat_record = RepeatRecord(
             domain=self.domain,
             payload_id='abc123',
             registered_at=datetime.utcnow(),
-            repeater_id=self.repeater.repeater_id,
+            repeater_id=repeater.repeater_id,
         )
 
         _process_repeat_record(repeat_record)
 
-        fetched_repeat_record = RepeatRecord.objects.get(id=repeat_record.id)
-        self.assertEqual(fetched_repeat_record.state, State.Cancelled)
-        self.assertEqual(self.mock_fire.call_count, 0)
-        self.assertEqual(self.mock_postpone_by.call_count, 0)
+        self.assertEqual(self.mock_fire.call_count, 1)
 
-    def test_cancels_and_returns_if_repeat_record_exceeds_max_retries(self):
-        repeat_record = RepeatRecord(
-            domain=self.domain,
-            payload_id='abc123',
-            registered_at=datetime.utcnow(),
-            repeater_id=self.repeater.repeater_id,
-            state=State.Fail,
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = 'process-repeat-record-tests'
+        cls.conn_settings = ConnectionSettings.objects.create(
+            domain=cls.domain,
+            name='To Be Deleted',
+            url="http://localhost/api/"
+        )
+        cls.repeater = Repeater.objects.create(
+            domain=cls.domain,
+            connection_settings=cls.conn_settings,
         )
 
-        with patch.object(RepeatRecord, "num_attempts", 1), \
-                patch.object(repeat_record, "max_possible_tries", 1):
-            _process_repeat_record(repeat_record)
+    def setUp(self):
+        self.patch()
 
-        fetched_repeat_record = RepeatRecord.objects.get(id=repeat_record.id)
-        self.assertEqual(fetched_repeat_record.state, State.Cancelled)
-        self.assertEqual(self.mock_fire.call_count, 0)
-        self.assertEqual(self.mock_postpone_by.call_count, 0)
+    def patch(self):
+        patch_fire = patch.object(RepeatRecord, 'fire')
+        self.mock_fire = patch_fire.start()
+        self.addCleanup(patch_fire.stop)
 
-    def test_deletes_repeat_record_cancels_and_returns_if_repeater_deleted(self):
-        deleted_repeater = Repeater.objects.create(
-            domain=self.domain,
-            connection_settings=self.conn_settings,
-            is_deleted=True
-        )
 
-        repeat_record = RepeatRecord(
-            domain=self.domain,
-            payload_id='abc123',
-            registered_at=datetime.utcnow(),
-            repeater_id=deleted_repeater.repeater_id,
-        )
+class TestIterReadyRepeaters(TestCase):
```

Comment:
I might be missing something, but can this not be a `SimpleTestCase`? I don't see any SQL objects saved anywhere.

Reply:
Good call. `Repeater.rate_limit()` was making a query, but I've patched that in 3d4e4cd89fd and made the test a `SimpleTestCase`.